### DISCUSSION TITLE
[Question][Hw7] How to get callee function name in call opcode instruction?

### DISCUSSION THREAD

**Discussion by sjoon2455** [prosyslab-classroom/cs524-program-analysis#368] [Q&A]
Hi, I would like to extract ```f``` from ```%call = call i32 @f(i32 %7)```.
I tried:
1. ```Llvm.operand instr 0```  -> output: %7
2. ```Llvm.value_name instr```  -> output: call

What is an appropriate API to get ```f```?
Thanks in advance(_ _)

**Comment by mekty2012** [comment]
You can use last element of Llvm.operand instr.
Check following source code of Llvmutils.
https://github.com/prosyslab-classroom/llvmutils/blob/df744f72deef6a9af133c8e0920c4c0babfc9de9/src/llvmutils.ml#L143

**Comment by sjoon2455** [comment]
@mekty2012 Thank you! I forgot to check Llvmutils 😅 Thanks again!

**Comment by sjoon2455** [comment]
@mekty2012, I have a question. Can you call ```Llvmutils.function_name``` in a code without error?

**Comment by mekty2012** [comment]
@sjoon2455 I had no problem when using it, but I got similar issue before.
Are you having segmentation fault? If so, it can happen when you are calling it to llvalue that is not contained in some function, like a constant integer. 
(It will return Null for such llvalue, then it will create segfault later.)

**Comment by sjoon2455** [comment]
@mekty2012 Not even a segmentation fault. If I type ```Llvmutils.function_name```, it says not defined with red line.
Since I can just implement it inline, it's not a big deal!
Thanks!
